pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/ModularLoadManagerImpl.java

```diff
@@ -1131,11 +1161,16 @@ public void writeBundleDataOnZooKeeper() {
         // Write the bundle data to metadata store.
         List<CompletableFuture<Void>> futures = new ArrayList<>();
 
-        for (Map.Entry<String, BundleData> entry : loadData.getBundleData().entrySet()) {
-            final String bundle = entry.getKey();
-            final BundleData data = entry.getValue();
-            futures.add(
-                    pulsarResources.getLoadBalanceResources().getBundleDataResources().updateBundleData(bundle, data));
+        // use synchronized to protect bundleArr.
+        synchronized (this) {
```

Comment:
Should you use `synchronized (bundleArr)`?

Reply:
good idea.